test/built-ins/TypedArray/prototype/find/predicate-may-detach-buffer.js

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.find
+es6id: 22.2.3.10
+description: >
+  Predicate may detach the buffer
+info: >
+  22.2.3.10 %TypedArray%.prototype.find (predicate [ , thisArg ] )
+
+  %TypedArray%.prototype.find is a distinct function that implements the same
+  algorithm as Array.prototype.find as defined in 22.1.3.8
+
+  ...
+
+  However, such optimization must not introduce any observable changes in the
+  specified behaviour of the algorithm and must take into account the
+  possibility that calls to predicate may cause the this value to become
+  detached.
+
+  ...
+
+  22.1.3.8 Array.prototype.find ( predicate[ , thisArg ] )
+
+  ...
+  4. If thisArg was supplied, let T be thisArg; else let T be undefined.
+  5. Let k be 0.
+  6. Repeat, while k < len
+    ...
+    b. Let kValue be ? Get(O, Pk).
+    c. Let testResult be ToBoolean(? Call(predicate, T, « kValue, k, O »)).
+  ...
+
+  9.4.5.8 IntegerIndexedElementGet ( O, index )
+
+  ...
+  3. Let buffer be the value of O's [[ViewedArrayBuffer]] internal slot.
+  4. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA(2);
+  var loops = 0;
+
+  assert.throws(TypeError, function() {
+    sample.find(function() {
+      loops++;
+      $DETACHBUFFER(sample.buffer);
+    });
+  }, "throws a TypeError getting a value from the detached buffer");
```

Comment:
I don't mean to say that throwing an error is logical. After all, most of the
tests we write are intended to ensure that implementors are not doing something
illogical. We have to be particularly careful with `$DETACHBUFFER` because it
"wraps" non-standard and under-specified functionality, so its behavior is
highly variable.